### PR TITLE
devtools as optional permission

### PR DESCRIPTION
**Pull request by rebloor**
#### Summary

Addresses the documentation needs of [Bug 1606862](https://bugzilla.mozilla.org/show_bug.cgi?id=1606862) Make devtools an optional permission.

No reference found to this permission in the Chrome documentation e.g. https://developer.chrome.com/docs/extensions/mv3/declare_permissions/#permissions or https://developer.chrome.com/docs/extensions/reference/devtools_inspectedWindow/

Confirmed as included in Safari on this page: https://developer.apple.com/documentation/safariservices/safari_web_extensions/adding_a_web_development_tool_to_safari_web_inspector#3966899

### CODE REVIEW

File: webextensions/manifest/optional_permissions.json [hunk 1]
```diff
@@ -333,6 +333,26 @@
...
+            "description": "<code>devtools</code>",
+            "support": {
+              "chrome": {
+                "version_added": false
+              },
+              "edge": "mirror",
+              "firefox": {
+                "version_added": "77"
+              },
+              "firefox_android": "mirror",
```
Comment (Rob--W) [inline webextensions/manifest/optional_permissions.json:347]:
```suggestion
              "firefox_android": {
                "version_added": false
              },
```

The `devtools_page` feature and the associated `devtools` permission are desktop-only.

File: webextensions/manifest/permissions.json [hunk 2]
```diff
@@ -413,6 +413,26 @@
...
+            "description": "<code>devtools</code>",
+            "support": {
+              "chrome": {
+                "version_added": false
+              },
+              "edge": "mirror",
+              "firefox": {
+                "version_added": "54"
+              },
+              "firefox_android": "mirror",
```
Comment (Rob--W) [inline webextensions/manifest/permissions.json:427]:
```suggestion
              "firefox_android": {
                "version_added": false
              },
```

The `devtools_page` feature and the associated `devtools` permission are desktop-only.

File: webextensions/manifest/permissions.json [hunk 3]
```diff
@@ -413,6 +413,26 @@
...
+        "devtools": {
+          "__compat": {
+            "description": "<code>devtools</code>",
+            "support": {
+              "chrome": {
+                "version_added": false
+              },
+              "edge": "mirror",
+              "firefox": {
+                "version_added": "54"
```
Comment (Rob--W) [inline webextensions/manifest/permissions.json:425]:
```suggestion
                "version_added": "77"
```

The permission was introduced in version 77 as a consequence of supporting optional permissions.

Before Firefox 77 (including back to 54) and in Chromium browsers, the devtools namespace is available when devtools_page is used. Would it make sense to add a note to emphasize that the permission is implicitly granted when `devtools_page` is present? Or does the mdn/content addition suffice?

**Review by Rob--W — CHANGES_REQUESTED** [review]

File: webextensions/manifest/permissions.json (hunk 3 above)
Comment (rebloor) [reply webextensions/manifest/permissions.json]: > Would it make sense to add a note to emphasize that the permission is implicitly granted when `devtools_page` is present? Or does the mdn/content addition suffice?

I think that the details in the mdn/content addition should provide enough information without the need to repeat as a BCD note.

**Review by Rob--W — APPROVED** [review]

**Pull request merged**